### PR TITLE
Unify yaml serialization dependencies

### LINKED ISSUE #256: Consider updating official kubectl plugins doc with krew
Now that Krew is officially under kubernetes-sigs umbrella, it might be worthy of our time to update the Kubectl Plugins documentation at https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/ so that developers creating plugins know an easy way to package and distribute them.

/kind documentation
/priority P2

### LINKED ISSUE #265: Eliminate different yaml parsing/serializing dependencies
Looks like we ended up with 3 different yaml parses in this repo (oops).

1. `k8s.io/apimachinery/pkg/util/yaml` (originally used) used in `pkg/index/indexscanner` to parse plugin manifest files
2. `gopkg.in/yaml.v2` used by `cmd/validate-krew-manifest`
3. `github.com/ghodss/yaml` used in pkg/receipt and pkg/info (added in #195)

Ideally, I'd be eliminating number 2 & 3. However, it looks like `k8s.io/apimachinery/pkg/util/yaml` actually doesn't know how to serialize yaml (no marshal method).

Looks like (https://godoc.org/k8s.io/apimachinery/pkg/util/yaml?import-graph&hide=1) that package is depending on `sigs.k8s.io/yaml` which depends on `gopkg.in/yaml.v2`. So I'll probably just depend on `sigs.k8s.io/yaml`  instead.

/kind cleanup

### PR DESCRIPTION
**Pull request by ahmetb**
- Standardize on sigs.k8s.io/yaml.
-  Bump k8s.io/apimachinery depenceny as it used to depend on module    `github.com/ghodss/yaml`

Fixes #265.

/hold
Currently there are some test issues. Looks like the parser is actually doing
some unnecessary defaulting on metav1.ObjectMeta fields. This is making tests
difficult.

### CODE REVIEW

**Comment by ahmetb** [comment]
Hmm interesting enough this test was failing locally at some point (like below) but they're good now.

```
--- FAIL: TestStore (0.01s)
    receipt_test.go:71:   &index.Plugin{
          	TypeMeta: v1.TypeMeta{Kind: "Plugin", APIVersion: "krew.googlecontainertools.github.com/v1alpha2"},
          	ObjectMeta: v1.ObjectMeta{
          		... // 8 identical fields
          		DeletionTimestamp:          nil,
          		DeletionGracePeriodSeconds: nil,
        - 		Labels:                     nil,
        + 		Labels:                     map[string]string{},
        - 		Annotations:                nil,
        + 		Annotations:                map[string]string{},
        - 		OwnerReferences:            nil,
        + 		OwnerReferences:            []v1.OwnerReference{},
        - 		Finalizers:                 nil,
        + 		Finalizers:                 []string{},
          		ClusterName:                "",
        - 		ManagedFields:              nil,
        + 		ManagedFields:              []v1.ManagedFieldsEntry{},
          	},
          	Spec: index.PluginSpec{ShortDescription: "short", Platforms: []index.Platform{{URI: "http://example.com", Sha256: "deadbeef", Files: []index.FileOperation{{}}, Bin: "foo"}}},
          }

FAIL
FAIL	sigs.k8s.io/krew/pkg/receipt	0.785s
```

File: pkg/index/indexscanner/scanner.go [hunk 1]
```diff
@@ -86,7 +84,6 @@ func LoadPluginFileFromFS(pluginsDir, pluginName string) (index.Plugin, error) {
 
 // ReadPluginFile loads a file from the FS. When plugin file not found, it
 // returns an error that can be checked with os.IsNotExist.
-// TODO(lbb): Add object verification
```
Comment (corneliusweig) [inline pkg/index/indexscanner/scanner.go]: :tada:

**Comment by ahmetb** [comment]
Done.

File: pkg/index/indexscanner/scanner.go (hunk 1 above)
Comment (thecraftman) [reply pkg/index/indexscanner/scanner.go]: nice

Comment (ahmetb) [reply pkg/index/indexscanner/scanner.go]: ?

**Pull request merged**